django/apps/registry.py

```diff
@@ -208,6 +209,7 @@ def register_model(self, app_label, model):
                     (model_name, app_label, app_models[model_name], model))
         app_models[model_name] = model
         self.clear_cache()
+        self.do_pending_operations(model)
```

Comment:
I don't understand why you need to resolve all dependencies once you cleared the cache.